include/graphblas/reference/compressed_storage.hpp

```diff
@@ -501,6 +501,39 @@ namespace grb {
 					return 2 * nz + m + 1;
 				}
 
+				template< typename ValueType >
+				void assignValue(
+					const size_t nz, const size_t,
+					const size_t start, size_t end,
+					const ValueType& val
+				) {
+					// The value can not be of type void
+					static_assert(
+						!std::is_void< ValueType >::value,
+						"ValueType must not be void"
+					);
+					// The value must be convertible to D
+					static_assert(
+						std::is_convertible< ValueType, D >::value,
+						"ValueType must be convertible to D"
+					);
```

Comment:
This is a correct assertion, but for readability of compiler errors, the code standard is to put these static_asserts right at the beginning of each possible entry point (meaning each backend implementation of set should have a static assert like this)